### ISSUE TITLE
Edges of dataset load black patches

### ISSUE THREAD

**Issue by Modexus** [microsoft/torchgeo#548]
As all the tiles are warped to one `crs` I believe the edge of the `dataset` loads black `patches`.
I attached an example of a warped `NAIP` tile that is rotated because of the warping.
The `BoundingBox` is based on min & max values therefore capures more than the actual outline of the `tile`.

<img src="https://user-images.githubusercontent.com/37351874/171167579-c650f014-5aeb-474e-871e-de6eec66807b.png" width="200" />

Is there a good way to handle this? 
Maybe shrinking the `BoundingBox` somehow to only contain data.

**Comment by adamjstewart** [comment]
I haven't yet found a good way to handle this. If we shrink the bounds of the tile, we lose valuable information in the corners that do reach the edge. For grid tiles like NAIP the nodata values in the corner should be replaced by the values of neighboring tiles. The real issue is satellites like Landsat/Sentinel where we don't have this.

**Comment by Modexus** [comment]
That is true; the `NAIP` tiles should actually fully contain my `roi` - I probably handle something incorrectly.

I would think that losing the corner information is not that important in most datasets, just less data.
The option to use the smaller bounding box would be a good addition as it is more general than handling nodata value in each trainer.

In the trainer one option I'd see is to filter the nodata values which in turn corresponds to using the smaller bounding box.

**Comment by adamjstewart** [comment]
It depends on if your goal is training or prediction. For training, yes, it's fine to lose a bit of data assuming you have a lot of data. For prediction, you want to make a prediction for every pixel on the map, not just the ones near the center of each image tile. I agree that this would be a useful addition though. I'm not sure if there's a way to automatically predict the maximum size of the bounding box within the image that contains data. I guess it depends on the metadata available for that particular satellite.

**Comment by Modexus** [comment]
Ah yes predicting, I didn't think of that as I only really train for research.

Are the landsat images already rotated or is it the same effect from warping?
It may be feasible to warp the coordinates of the bounds and use the smaller value for the bounding box.

If they are already rotated is it possible to warp the bounds back to the original crs (if known) and then warp the coordinates to the new crs?
